homeassistant/components/aurora_abb_powerone/aurora_client.py

```diff
@@ -0,0 +1,136 @@
+"""Abstraction over aurorapy client to simplify usage in Home Assistant."""
+
+import contextlib
+from dataclasses import dataclass
+
+from aurorapy.client import (
+    AuroraBaseClient,
+    AuroraError,
+    AuroraSerialClient,
+    AuroraTCPClient,
+    AuroraTimeoutError,
+)
+from serial import SerialException
+
+
+@dataclass
+class AuroraInverterIdentifier:
+    """Identifier information about the inverter."""
+
+    serial_number: str
+    model: str
+    firmware: str
+
+
+@dataclass
+class AuroraInverterData:
+    """Data (measurements) read from the inverter."""
+
+    grid_voltage: float
+    grid_current: float
+    instantaneouspower: float
+    grid_frequency: float
+    i_leak_dcdc: float
+    i_leak_inverter: float
+    temp: float
+    r_iso: float
+    totalenergy: float
+    alarm: str
+
+
+class AuroraClientTimeoutError(Exception):
+    """Timeout error specific to AuroraClient."""
+
+
+class AuroraClientError(Exception):
+    """Generic error specific to AuroraClient."""
+
+
+class AuroraClient:
+    """Abstracts aurorapy integration calls."""
+
+    _client: AuroraSerialClient | AuroraTCPClient
+
+    def __init__(self, client: AuroraBaseClient) -> None:
+        """Initialize the AuroraClient with a given aurorapy client."""
+        self._client = client
+
+    @classmethod
+    def from_serial(cls, inverter_serial_address: int, serial_comport: str):
```

Comment:
Can you add the type decorator for the return type?
```suggestion
    def from_serial(cls, inverter_serial_address: int, serial_comport: str) -> AuroraClient:
```